### PR TITLE
Adding Typescript and Mantine example

### PR DESCRIPTION
**Pull request by chaqchase**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: examples/with-typescript-mantine/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# TypeScript & Mantine Next.js example
+
+This is a really simple project that show the usage of Next.js with TypeScript and Mantine.
```
Comment (balazsorban44) [inline examples/with-typescript-mantine/README.md:3]:
```suggestion
This is a simple project that shows the usage of Next.js with TypeScript and Mantine.
```

File: examples/with-typescript-mantine/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+pnpm create next-app -- --example with-typescript-mantine with-typescript-mantine-app
+```
+
+Deploy it to the cloud with [Vercel](https://vercel.com/new?utm_source=github&utm_medium=readme&utm_campaign=next-example) ([Documentation](https://nextjs.org/docs/deployment)).
+
+## Notes
+
+This is an amalgamation of the 1 existing example:
+
+- [with-typescript](https://github.com/vercel/next.js/tree/canary/examples/with-typescript)
```
Comment (balazsorban44) [inline examples/with-typescript-mantine/README.md:29]:
```suggestion
```

I don't think this is a useful note, let's remove it.

File: examples/with-typescript-mantine/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@mantine/core": "^4.1.4",
+    "@mantine/hooks": "^4.1.4",
+    "@mantine/next": "^4.1.4",
+    "next": "12.1.5",
```
Comment (balazsorban44) [inline examples/with-typescript-mantine/package.json:13]:
```suggestion
    "next": "latest",
```

File: examples/with-typescript-mantine/pages/api/hello.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
```
Comment (balazsorban44) [inline examples/with-typescript-mantine/pages/api/hello.ts:1]: This file does not seem to showcase Mantine, I would remove it to keep the example focused on its goal.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks! I added some comments.

I think we can just call the folder `with-mantine` and drop `-typescript`, as we plan to only add TS examples whenever possible, as mentioned in https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples

**Review by balazsorban44 — DISMISSED** [review]
Thanks!

**Comment by triyanox** [comment]
I have made some changes to pass all the test including the failing linting test due to code formatting !

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**